### DISCUSSION TITLE
search feature - if you want it

### DISCUSSION THREAD

**Discussion by jgarza9788** [Schneegans/Burn-My-Windows#462] [Ideas]
https://github.com/jgarza9788/Burn-My-Windows/tree/searchfeature

* re-organized the top widgets ... to make it look nice
* added the search field
* added a filter function (it's simple, but it will do for now) (maybe something like a fuzzy search later)
* made a description field for each effect - so things like "flame" will work for fire and incinerate

![image](https://github.com/user-attachments/assets/4a71c382-670c-4019-b36c-16b578dc186a)

**Comment by Schneegans** [comment]
I think this is a good idea. Especially if you want to add more effects in the future :smile: 

Although I am not sure about the descriptions - these are very long and pretty hard to translate. This will make it much harder to translate the project and will only bring a small benefit.

And maybe the "All" and "None" buttons should be renamed "Enable All" and "Disable All" to make clear that they do not belong to the search bar.

What do you think?

**Comment by jgarza9788** [comment]
Yup, I was also thinking about … a filter for enabled and disabled effects.

I like working with you, so you can expect this soon… just keep in mind I
have a 9-5 and I gotta get some time in for sun and fresh air.

**Comment by jgarza9788** [comment]
i don't mind if you use it or not.
but here it is with the button text changed.

it only makes sense if there are a lot more effects,
we'll see if that happens... i only have a few more in mind to try and submit.

![image](https://github.com/user-attachments/assets/e0e19317-23c3-48a0-9e2c-15787ecdc633)

i also tried out filter buttons (i.e. only show enabled, only show disabled, and all)
![image](https://github.com/user-attachments/assets/f1cbde88-7434-46d3-9bab-531a2fc26d8a)
![image](https://github.com/user-attachments/assets/995c3f2c-9248-43fe-bb5c-38d348989ce5)

... 

if was a fun little experiment in UI.
but i'll focus more on effects now unless you want something specific with the UI

**Reply by Schneegans** [reply]
I think the variant with a search bar and the two buttons is great. Do you want to propose this as a pull request?

**Reply by jgarza9788** [reply]
Sure, I can make a clean branch for you (maybe in a few days, work is getting a bit busy)
